invenio_app_ils/indexer.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2018 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Ils Records API."""
+
+from __future__ import absolute_import, print_function
+
+import logging
+from datetime import datetime, timedelta
+from flask import current_app
+from celery import shared_task
+from invenio_app_ils.circulation.utils import circulation_document_retriever, \
+    circulation_items_retriever
+from invenio_app_ils.records.api import Document, Item
+from invenio_circulation.api import is_item_available, Loan
+from invenio_circulation.search.api import search_by_pid
+from invenio_indexer.api import RecordIndexer
+from invenio_jsonschemas import current_jsonschemas
+from invenio_pidstore.resolver import Resolver
+from invenio_records.api import Record
+
+
+@shared_task(ignore_result=True)
+def index_loans_after_item_indexed(item_pid):
+    """Index loan to refresh item reference."""
+    loan_search = search_by_pid(item_pid=item_pid)
+    loan_ids = []
+    for loan in loan_search.scan():
+        record = Loan.get_record_by_pid(loan[Loan.pid_field])
+        if record:
+            loan_ids.append(record.id)
+
+    RecordIndexer().bulk_index(loan_ids)
+
+
+@shared_task(ignore_result=True)
+def index_document_after_item_indexed(item_pid):
+    """Index document to re-compute circulation information."""
+    document_pid = circulation_document_retriever(item_pid)
+    document = Document.get_record_by_pid(document_pid)
+    if document:
+        RecordIndexer().index(document)
+    logging.debug("Document indexed: {0}".format(document_pid))
+
+
+class ItemIndexer(RecordIndexer):
+    """Indexer class for `Item`."""
+
+    def index(self, item):
+        """Index an item."""
+        super(ItemIndexer, self).index(item)
+        index_loans_after_item_indexed.apply_async(
+            (item[Item.pid_field],),
+            eta=datetime.utcnow() + timedelta(seconds=5),
```

Comment:
Would it be possible to have this seconds in a config? So we can tune it in qa/prod?

Reply:
Sure!

Reply:
Added in the issue of fixing resolvers.